src/ansys/fluent/core/quantity.py

```diff
@@ -0,0 +1,601 @@
+
+                self._si_multiplier *= get_si_conversion_factor(unit_str) ** term_power
+
+            elif unit_str in _UnitsTable.derived_units_with_conversion_factor:
+                (
+                    conversion_factor,
+                    unit_str,
+                ) = _UnitsTable.derived_units_with_conversion_factor[unit_str]
+                self._si_multiplier *= conversion_factor**term_power
+                self._computemultipliers_and_offsets(unit_str, term_power)
+            elif unit_str in _UnitsTable.derived_units:
+                self._computemultipliers_and_offsets(
+                    _UnitsTable.derived_units[unit_str], term_power
+                )
+
+
+class Dimension(object):
+    def __init__(self, unit_str):
+        self._dimensions = {
+            "M": 0.0,
+            "L": 0.0,
+            "T": 0.0,
+            "Temp": 0.0,
+            "I": 0.0,
+            "N": 0.0,
+            "J": 0.0,
+            "Angle": 0.0,
+            "SAngle": 0.0,
+            "": 0.0,
+        }
+        self._parser(unit_str)
+
+    def _add_or_update_dimension(self, dim_dict):
+        for key in dim_dict.keys():
+            self._dimensions[key] += dim_dict[key]
+
+    def _parser(self, unit, power=1):
+        if len(unit) == 0:
+            return
+
+        unit_list = unit.split(" ")
+        for term in unit_list:
+            unit_dim = ""
+            has_multiplier = not (
+                term in _UnitsTable.fundamental_units
+                or term in _UnitsTable.derived_units
+            )
+            unit_str = remove_multiplier(term) if has_multiplier else term
+            term_power = 1
+
+            if "^" in term:
+                unit_str, term_power = term[: term.index("^")], float(
+                    term[term.index("^") + 1 :]
+                )
+                has_multiplier = not (
+                    unit_str in _UnitsTable.fundamental_units
+                    or unit_str in _UnitsTable.derived_units
+                )
+                unit_str = remove_multiplier(unit_str) if has_multiplier else unit_str
+
+            term_power *= power
+            unit_dim = self._get_dim(unit_str, term_power)
+
+            if unit_dim != None:
+                self._add_or_update_dimension(unit_dim)
+
+    def as_dict(self):
+        return self._dimensions
+
+    def _get_dim(self, unit_str, power):
+        if unit_str in _UnitsTable.fundamental_units:
+            return {_UnitsTable.fundamental_units[unit_str]: power}
+        elif unit_str in _UnitsTable.derived_units:
+            self._parser(_UnitsTable.derived_units[unit_str], power)
+        elif unit_str in _UnitsTable.derived_units_with_conversion_factor:
+            _, unit_str = _UnitsTable.derived_units_with_conversion_factor[unit_str]
+            self._parser(unit_str, power)
+        else:
+            raise ValueError("Not implemented")
+
+
+def get_si_unit_from_dim(dim_list):
+    si_unit = ""
+    dim_to_unit_map = {
```

Comment:
If you restructure `UnitSystem` as I suggested, then you can get this information from there I think.

Reply:
Thank you very much @dnwillia-work .